### PR TITLE
add champ conda-forge recipe 

### PR DESCRIPTION
**Pull request by wweir827**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by wweir827** [comment]
one of the dependencies, igraph is not available for windows build, and hence the checks are failing for win_64.  Thanks

**Comment by wweir827** [comment]
@conda-forge/staged-recipes I think this should be good to go.  Thanks.

File: recipes/champ/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+build:
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  number: 0
+  skip: True  # [win]
+
+
+requirements:
+  build:
+    - pkg-config
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: Does this contain any compiled code? I don't see it in the upstream repository?

Comment (wweir827) [reply recipes/champ/meta.yaml]: the champ package doesn't contain any compiled code.  one of the dependencies does (louvain ) so I had included that in there .  Can I remove it?  Thanks.

File: recipes/champ/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  version: {{ version }}
+
+source:
+  url: https://files.pythonhosted.org/packages/ce/25/b7cba31f1781a7ddbcafb8ff8e96357ffa98d41704ccf7e19bcfe928649a/champ-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  number: 0
+  skip: True  # [win]
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: It's fine if this makes the Windows build fail:
```suggestion
  noarch: python
```

File: recipes/champ/meta.yaml (hunk 1 above)
Comment (chrisburr) [reply recipes/champ/meta.yaml]: Yes you can remove all of the `build` dependencies. Each package only cares about its direct dependencies.

File: recipes/champ/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{% set name = "champ" %}
+{% set version = "2.0.2" %}
+{% set sha256 = "2e24d8fea58bf65bb13ec646cbb7404c4fc780ca4c07d5eb03c9aca9cb1609bb" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://files.pythonhosted.org/packages/ce/25/b7cba31f1781a7ddbcafb8ff8e96357ffa98d41704ccf7e19bcfe928649a/champ-{{ version }}.tar.gz
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: Can you change this with a predicable URL like the example has

File: recipes/champ/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - seaborn
+    - tqdm
+    - louvain
+    - scikit-learn
+    - future
+
+
+test:
+  imports:
+    - igraph
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: I think the only import test should be champ? Or am I missing something?
```suggestion
    - champ
```

File: recipes/champ/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    - louvain
+
+about:
+  home: https://github.com/wweir827/CHAMP
+  license: MIT
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]:
```suggestion
  license: MIT
  license_family: MIT
  license_file: LICENSE
```
Ideally the licence file should be included in the PyPI tarball by listing it in the `MANIFEST.in` file. As this isn't currently the case you can add it to the recipe directory instead.

File: recipes/champ/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+
+  host:
+    - python
+    - igraph
+    - setuptools
+
+  run:
+    - python-igraph
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]:
```suggestion
    - python
    - python-igraph
```

File: recipes/champ/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://files.pythonhosted.org/packages/ce/25/b7cba31f1781a7ddbcafb8ff8e96357ffa98d41704ccf7e19bcfe928649a/champ-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: Can you replace this with the command in the example

Comment (wweir827) [reply recipes/champ/meta.yaml]: done

File: recipes/champ/meta.yaml (hunk 3 above)
Comment (wweir827) [reply recipes/champ/meta.yaml]: changed url

File: recipes/champ/meta.yaml (hunk 1 above)
Comment (wweir827) [reply recipes/champ/meta.yaml]: removed all build dependencies

File: recipes/champ/meta.yaml (hunk 5 above)
Comment (wweir827) [reply recipes/champ/meta.yaml]: added license file to folder with recipe

File: recipes/champ/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  number: 0
+  noarch: python
+
+
+requirements:
+
+  host:
+    - python
+    - igraph
+    - setuptools
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: You need pip here

File: recipes/champ/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    - tqdm
+    - louvain
+    - scikit-learn
+    - future
+
+
+test:
+  imports:
+    - champ
+    - louvain
```
Comment (chrisburr) [inline recipes/champ/meta.yaml]: This is coming from a dependency right? Or does `champ` effect how it is imported?
```suggestion
```

Comment (wweir827) [reply recipes/champ/meta.yaml]: champ doesn't affect import. removed from test.  thanks.

**Pull request merged**